app/controllers/home_controller.rb

```diff
@@ -1,5 +1,125 @@
 class HomeController < ApplicationController
   before_action :authenticate_user!
   def index
+    @id = current_user.email
   end
+
+  def getYears
+    sems = Course.where(teacher:@id)
+    uniqSems = Set.new
+    sems.each do |s|
+      uniqSems << s.semester[-4..-1]
+    end
+    return uniqSems.length()
+  end
+  helper_method :getYears
+
+
+  def getAvgTotal
+    avg = Quiz.where(teacher:@id).average(:score)
+    if avg.nil?
+      avg = "No quizzes taken!"
+    else 
+      avg = avg.round(2)
+    end
+
+    strk = Quiz.where(teacher:@id).average(:longest_streak)
+    if strk.nil?
+      strk = "No quizzes taken!"
+    else
+      strk = strk.round(2)
+    end
+
+    return [avg, strk]
+  end
+  helper_method :getAvgTotal
+
+  def getAvgRecent
+    recentFall = Course.where(teacher:@id).where("semester like ?", "%#{"Fall"}%").order(semester: :desc).first
+    fallYear = (recentFall.semester[-4..-1] || "0").to_i
+    recentSpring = Course.where(teacher:@id).where("semester like ?", "%#{"Spring"}%").order(semester: :desc).first
```

Comment:
I think we should also include Summer

Reply:
Ask prof Summer I, Summer II, Winter semester, etc

Reply:
Holding off until we know syntax/naming scheme

Reply:
Sounds good.